### PR TITLE
feat(extension-chrome): support send transaction 

### LINKED ISSUE #217: Support for the `ckb_sendTransaction` in Nexus
## Motivation

Due to security considerations, the dApp cannot access the entry of the CKB node that Nexus is connected to, so the dApp can only get the currently connected network according to `ckb_getBlockChainInfo` to determine which network should broadcast the transaction, which makes the dApp development inconvenient, we need to consider support for sendTransaction

## How to

https://github.com/ckb-js/nexus/blob/b8a62051c9766282e363a634c8def017b4460080/packages/extension-chrome/src/rpc/ckbImpl.ts#L3-L6

### PR DESCRIPTION
**Pull request by IronLu233**
# What Changed
This method add new RPC method named `ckb_sendTransaction`
resolve #217 
## Motivation
Give an interface to send transaction to current network for DApps.
## Change Type

Indicate the type of change your pull request is:

- [ ] `documentation`
- [ ] `patch`
- [x] `minor`
- [ ] `major`

:baby_chick: Download canary assets:

[nexus--canary.227.4777303604.zip](https://github.com/ckb-js/nexus/releases/download/v0.0.0-canary/nexus--canary.227.4777303604.zip)

<details>
  <summary>📦 Published PR as canary version: <code>0.0.16--canary.227.4777303604.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @nexus-wallet/ownership-providers@0.0.16--canary.227.4777303604.0
  npm install @nexus-wallet/protocol@0.0.16--canary.227.4777303604.0
  npm install @nexus-wallet/utils@0.0.16--canary.227.4777303604.0
  # or 
  yarn add @nexus-wallet/ownership-providers@0.0.16--canary.227.4777303604.0
  yarn add @nexus-wallet/protocol@0.0.16--canary.227.4777303604.0
  yarn add @nexus-wallet/utils@0.0.16--canary.227.4777303604.0
  ```
</details>

### CODE REVIEW

**Comment by homura** [comment]
could you please help to fix the bug of RPC docs

https://github.com/ckb-js/nexus/blob/a2a619a3e6b218ff64fc4cf1b05243fca72a99e7/package.json#L80

```diff
- "packages/types/**/*": [
+ "packages/protocol/**/*": [
```

File: packages/protocol/src/ckb.ts [hunk 1]
```diff
@@ -30,4 +30,17 @@ export interface CkbMethods {
...
+   * @returns transaction hash of the sent transaction
+   * @throws when the transaction is invalid, or meets network issue
+   * @example
+   * ```ts
+   * const signedTransaction;// signed transaction with your business logic
+   * const txHash = await window.ckb.request({ method: "ckb_sendTransaction", params: { tx: signedTransaction } });
+   * // if your transaction is to mainnet or testnet, you can view it on https://explorer.nervos.org/transaction/${txHash}
+   * ```
+   */
+  ckb_sendTransaction(payload: { tx: Transaction }): Promise<string>;
```
Comment (homura) [inline packages/protocol/src/ckb.ts:45]:
```ts

type OutputsValidator = "passthrough" | "well_known_scripts_only";


function ckb_sendTransaction(payload: {
  transaction: Transaction,
  outputValidator?: OutputValidator,
): Promise<Hash>;
```

the `ckb_*` need to keep same with the [CKB](https://github.com/nervosnetwork/ckb/blob/develop/rpc/README.md#method-send_transaction)

**Comment by homura** [comment]
I doubt that this PR has really been tested, because the parameters of ckb `send_transaction `use the snake case, and the type of `ckb_sendTransaction` in this PR is not the same as CKB, so I understand that it is not possible to `sendtransaction` success

File: packages/extension-chrome/src/services/ownership/backend/index.ts [hunk 2]
```diff
@@ -173,7 +180,10 @@ export function createBackend(_payload: { nodeUrl: string }): Backend {
       return ResultFormatter.toBlockchainInfo(res);
     },
     sendTransaction(tx: Transaction) {
-      return client.request<Hash, Transaction>('send_transaction', tx);
+      return client.request<Hash, [RpcType.RawTransaction, 'well_known_scripts_only' | 'passthrough' | null]>(
+        'send_transaction',
+        [toRpcTransaction(tx), 'passthrough'],
```
Comment (homura) [inline packages/extension-chrome/src/services/ownership/backend/index.ts]: `ResultFormatter.toRawTransaction` is fine

Comment (IronLu233) [reply packages/extension-chrome/src/services/ownership/backend/index.ts]: Not have this method yet. 
image

And `ResultFormatter.toRawTransaction` received rpc transaction, return lumos transaction. it is reversed of expected.

Comment (homura) [reply packages/extension-chrome/src/services/ownership/backend/index.ts]: typo, we should use `paramsFormatter.toRawTransaction`

Comment (IronLu233) [reply packages/extension-chrome/src/services/ownership/backend/index.ts]: Yes, That make code more clear. and I replace other similar cases in this file.

**Review by homura — APPROVED** [review]
LGTM

**Pull request merged**